bridges/snowbridge/primitives/outbound-queue/src/v2/exporter.rs

```diff
@@ -0,0 +1,45 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: 2023 Snowfork <hello@snowfork.com>
+use core::marker::PhantomData;
+use snowbridge_core::operating_mode::IsHalted;
+use sp_std::vec::Vec;
+use xcm::{
+	prelude::{Location, SendError, SendResult, SendXcm, Xcm, XcmHash},
+	VersionedLocation, VersionedXcm,
+};
+use xcm_builder::InspectMessageQueues;
+
+pub struct PausableExporter<Halted, InnerExporter>(PhantomData<(Halted, InnerExporter)>);
+
+impl<Halted: IsHalted, InnerExporter: SendXcm> SendXcm for PausableExporter<Halted, InnerExporter> {
+	type Ticket = InnerExporter::Ticket;
+
+	fn validate(
+		destination: &mut Option<Location>,
+		message: &mut Option<Xcm<()>>,
+	) -> SendResult<Self::Ticket> {
+		match Halted::is_halted() {
```

Comment:
We can improve naming. Should not use "halted" within this context.

```suggestion
pub struct PausableExporter<PausedQuery, InnerExporter>(PhantomData<(Halted, InnerExporter)>);

impl<PausedQuery: ExportPausedQuery, InnerExporter: SendXcm> SendXcm for PausableExporter<Halted, InnerExporter> {
	type Ticket = InnerExporter::Ticket;

	fn validate(
		destination: &mut Option<Location>,
		message: &mut Option<Xcm<()>>,
	) -> SendResult<Self::Ticket> {
		match PausedQuery::is_paused() {
```

Reply:
https://github.com/vgeddes/polkadot-sdk/pull/7/commits/cf3aaac77798f7371b75f1a2832d579bd8ed37b2